cmd/snap-confine/snap-device-helper-test.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../libsnap-confine-private/test-utils.h"
+
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <string.h>
+
+// TODO: build at runtime
+static const char *sdh_path_default = "snap-confine/snap-device-helper";
+
+// A variant of unsetenv that is compatible with GDestroyNotify
+static void my_unsetenv(const char *k)
+{
+	g_unsetenv(k);
+}
+
+// A variant of rm_rf_tmp that calls g_free() on its parameter
+static void rm_rf_tmp_free(gchar * dirpath)
+{
+	rm_rf_tmp(dirpath);
+	g_free(dirpath);
+}
+
+static gchar *find_sdh_path(void)
+{
+	const char *sdh_from_env = g_getenv("SNAP_DEVICE_HELPER");
+	if (sdh_from_env != NULL) {
+		return g_strdup(sdh_from_env);
+	}
+	return g_strdup(sdh_path_default);
+}
+
+static int run_sdh(gchar * action,
```

Comment:
Indent looks a bit of here, but I guess that is what `make fmt` produced?

Reply:
Yeah. Not sure what's happening here, it seems to do the right thing for `char`. Maybe it treats unknown types differently?